server/src/main/java/org/elasticsearch/snapshots/InFlightShardSnapshotStates.java

```diff
@@ -102,7 +103,9 @@ private static boolean assertGenerationConsistency(
         @Nullable String activeGeneration
     ) {
         final String bestGeneration = generations.getOrDefault(indexName, Collections.emptyMap()).get(shardId);
-        assert bestGeneration == null || activeGeneration == null || activeGeneration.equals(bestGeneration);
+        if ((bestGeneration == null || activeGeneration == null || activeGeneration.equals(bestGeneration)) == false) {
+            throw new AssertionFailedException("gnarf");
```

Comment:
Changing this assertion failure into a runtime exception is, I think, cheating ;)

Reply:
sorry needed a breakpoint there :D